src/sentry/processing/realtime_metrics/redis.py

```diff
@@ -65,14 +66,13 @@ def increment_project_event_counter(self, project_id: int, timestamp: int) -> No
         in seconds since the UNIX epoch (i.e., as returned by time.time()).
         """
 
-        if self._counter_bucket_size > 1:
-            timestamp -= timestamp % self._counter_bucket_size
+        timestamp -= timestamp % self._counter_bucket_size
 
         key = f"{self._counter_key_prefix()}:{project_id}:{timestamp}"
 
         with self.cluster.pipeline() as pipeline:
             pipeline.incr(key)
-            pipeline.pexpire(key, self._counter_ttl)
+            pipeline.pexpire(key, int(self._counter_ttl / datetime.timedelta(milliseconds=1)))
```

Comment:
nit: i wonder if it would be less pythonic to use `int(self.counter_ttl.total_seconds() * 1000)` here? i'm assuming what you're trying to do here is to get the number of milliseconds in `_counter_ttl`.

Reply:
Yeah. I think I got this idiom from `datetime`'s documentation.

Reply:
since our buckets are a multiple of 10s, and you changed all the other variables to be in seconds, might as well use `expire` (with second resolution) instead of `pexpire` (ms resolution) here. I don’t think there is any benefit of having a ms resolution ttl.

Reply:
Actually you're totally right about that. Originally I made it so fine-grained for testing purposes, but all things considered that isn't a great reason.